File: Minimum Operations - GFG/minimum-operations.java
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;
class GfG
{
    public static void main(String args[])
        {
            Scanner sc = new Scanner(System.in);
            int t = sc.nextInt();
            while(t-->0)
                {
                    int n = sc.nextInt();
                    Solution ob = new Solution();
                    System.out.println(ob.minOperation(n));
                }
        }
}    
// } Driver Code Ends


//User function Template for Java


class Solution
{
    public int minOperation(int n)
    {
        int count = 0;
        while(n!=0)
            {
                // if n is even then it will be good to
                // reach n from n/2 by multiplying it by 2.
                if(n%2==0)
                    n/=2;
                // if n is odd then we can reach n from n-- only. 
                else
                    n--;
                
                count++;     
            }
        return count;
    }
}

File: Minimum Operations - GFG/minimum-operations.js
//{ Driver Code Starts
//Initial Template for javascript

'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.trim().split('\n').map(string => {
        return string.trim();
    });
    
    main();    
});

function readLine() {
    return inputString[currentLine++];
}

function main() {
    let t = parseInt(readLine());
    let i = 0;
    for(;i<t;i++)
    {
        let n = parseInt(readLine());
        let obj = new Solution();
        let res = obj.minOperation(n);
        if(res === -0)
            res = 0;
        console.log(res);
    }
}

// } Driver Code Ends


//User function Template for javascript


/**
 * @param {number} n
 * @return {number}
*/

class Solution {

    minOperation(n){ // Function to calculate the minimum number of operations required.
        let cnt = 0; // Initialize a variable to keep track of the number of operations.
        while(n > 0){ // Loop until the input number becomes 0.
            if(n%2){ // Check if the input number is odd.
                n--; // Decrement by 1 if the input number is odd.
               
            } else { // If the input number is even.
                n/=2; // Divide the input number by 2.
            }
            cnt++; // Increment the count of operations.
        }
        return cnt; // Return the total count of operations.
        
    }
}